ext/nnstreamer/tensor_filter/tensor_filter_tvm.cc

```diff
@@ -0,0 +1,436 @@
+const accl_hw tvm_subplugin::hw_list[] = { ACCL_CPU, ACCL_GPU };
+
+/**
+ * @brief Construct a new tvm subplugin::tvm subplugin object
+ */
+tvm_subplugin::tvm_subplugin ()
+  : tensor_filter_subplugin (), empty_model (true), model_path (nullptr),
+    device (DLDevice{kDLCPU, 0}), mod_factory (nullptr), gmod (nullptr)
+{
+  inputInfo.num_tensors = 0;
+  outputInfo.num_tensors = 0;
+}
+
+/**
+ * @brief Cleanup method for tvm subplugin
+ */
+void tvm_subplugin::cleanup () noexcept
+{
+  if (empty_model)
+    return;
+  if (model_path)
+    g_free(model_path);
+
+  input_info_list.clear ();
+  output_info_list.clear ();
+
+  model_path = nullptr;
+  inputInfo.num_tensors = 0;
+  outputInfo.num_tensors = 0;
+  empty_model = true;
+}
+
+/**
+ * @brief Destroy the tvm subplugin::tvm subplugin object
+ */
+tvm_subplugin::~tvm_subplugin ()
+{
+  cleanup ();
+}
+
+/**
+ * @brief Method to get an empty object
+ */
+tensor_filter_subplugin & tvm_subplugin::getEmptyInstance ()
+{
+  return *(new tvm_subplugin ());
+}
+
+/**
+ * @brief Internal method to parse custom properties
+ * @param custom_prop Given c_str value of 'custom' property,
+ *                    which contains device info
+ */
+bool tvm_subplugin::parse_custom_prop (const char *custom_prop)
+{
+  gchar **options;
+  bool invalid_option = false;
+
+  if (custom_prop == nullptr) {
+    /* no custom properties */
+    return true;
+  }
+
+  options = g_strsplit (custom_prop, ",", -1);
+  
+  for (guint op = 0; op < g_strv_length (options); ++op) {
+    gchar **option = g_strsplit (options[op], ":", -1);
+
+    if (g_strv_length (option) > 1) {
+      g_strstrip (option[0]);
+      g_strstrip (option[1]);
+
+      if (g_ascii_strcasecmp (option[0], "device") == 0) {
+        if (g_ascii_strcasecmp (option[1], "CPU") == 0) {
+          device = DLDevice{kDLCPU, 0};
+        } else if (g_ascii_strcasecmp (option[1], "GPU") == 0) {
+          device = DLDevice{kDLGPU, 0};
+        } else {
+          nns_loge ("Unknown device (%s).", option[1]);
+          invalid_option = true;
+        }
+      } else {
+        nns_logw ("Unknown option (%s).", options[op]);
+      }
+    }
+    if (invalid_option) {
+      g_strfreev (option);
+      break;
+    }
+    g_strfreev (option);
+  }
+  g_strfreev (options);
+  return !invalid_option;
+}
+
+/**
+ * @brief Configure tvm instance
+ */
+void tvm_subplugin::configure_instance (const GstTensorFilterProperties *prop)
+{
+  if (!parse_custom_prop (prop->custom_properties)) {
+    nns_loge ("Failed to parse custom property.");
+    cleanup();
+    throw std::invalid_argument ("Failed to parse custom property.");
+  }
+  
+  if (!empty_model) {
+    if (!prop->model_files[0] || prop->model_files[0][0] == '\0') {
+      std::cerr << "Model path is not given." << std::endl;
+      cleanup();
+      throw std::invalid_argument ("Model path is not given.");
+    }
+    cleanup ();
+  }
+
+  /* read model */
+  model_path = g_strdup (prop->model_files[0]);
```

Comment:
resource leak?

Reply:
IMO, `cleanup()` handles the allocated memory space. So it looks OK.